### PR TITLE
Changes to work with Mac OS X

### PR DESCRIPTION
**Pull request by nandub**
Made a few changes to make it work with Mac OS X

### CODE REVIEW

**Comment by claziss** [comment]
Anton, it is looking ok to me.

**Comment by nandub** [comment]
Is there anything else you want me to change please let me know.

BTW, so far I'm able to build the `arc-elf32` toolchain successful, but not able to test yet as I don't have anything to test it with.  For the `arc-uclibc` toolchain no such luck yet.

**Comment by jeremybennett** [comment]
nandup, don't forget the arc-elf32 toolchain includes a simulator, so you should be able to run programs from within GDB using "target sim", or call the simulator directly

```
arc-elf32-run <program>
```

Just remember it is a simple instruction set simulator, for testing the tool chain, not a full blown system simulator.

HTH,

**Comment by nandub** [comment]
@jeremybennett thanks, unfortunately I had to build the toolchain with `--no-sim` option, otherwise I get an error.  see https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/issues/36#issuecomment-30490902

File: build-all.sh [hunk 1]
```diff
@@ -327,7 +329,7 @@ case "x${DISABLE_MULTILIB}" in
 esac
 
 
-if [ x`uname -o` = "xMsys" ]
+if [ x`${UNAME} -o` = "xMsys" ]
```
Comment (anthony-kolesov) [inline build-all.sh]: Please add here check for MacOS as well. Thus simulator will not be built by default and there will be no need to pass "--no-sim" each time.

File: arc-init.sh [hunk 2]
```diff
@@ -85,6 +85,9 @@
 # $1 - build directory
 # $2 - tool to test (e.g. "binutils" will run "check-binutils"
 # $3 - log file
+
+. ${ARC_GNU}/toolchain/settings.sh
```
Comment (anthony-kolesov) [inline arc-init.sh]: I suggest to move code from "settings.sh" here. arc-init.sh looks like a good place to check for host platform.

File: build-elf32.sh [hunk 3]
```diff
@@ -165,13 +165,13 @@ then
     sim_build=all-sim
     # CGEN doesn't have install-strip target.
     sim_install=install-sim
-    sed -i "${ARC_GNU}"/gdb/gdb/configure.tgt \
-	-e 's!# gdb_sim=../sim/arc/libsim.a!gdb_sim=../sim/arc/libsim.a!'
+    ${SED} -i "${ARC_GNU}"/gdb/gdb/configure.tgt \
```
Comment (anthony-kolesov) [inline build-elf32.sh:168]: This file doesn't source settings.sh. If settings.sh will me merged to arc-init, than no changed is required.

File: build-kernel.sh [hunk 4]
```diff
@@ -60,6 +60,7 @@ ARC_UNPACKED_NAME=arc_initramfs_22_abi-v3-0.9.34
 d=`dirname "$0"`
 ARC_GNU=`(cd "$d/.." && pwd)`
 . "${ARC_GNU}"/toolchain/define-release.sh
+. "${ARC_GNU}"/toolchain/settings.sh
```
Comment (anthony-kolesov) [inline build-kernel.sh]: Should be arc-init.sh.

File: build-uclibc.sh [hunk 5]
```diff
@@ -282,30 +282,30 @@ make distclean >> "${logfile}" 2>&1 || true
 # uClibc 0.9.34 onwards use defconfig
 if [ ! -f extra/Configs/defconfigs/arc/defconfig ]
 then
-    sed -e "s#%KERNEL_HEADERS%#${tmp_install_dir}/include#" \
-        -e "s#%RUNTIME_PREFIX%#${tmp_install_dir}/${arche}-linux-uclibc/#" \
-        -e "s#%DEVEL_PREFIX%#${tmp_install_dir}/${arche}-linux-uclibc/#" \
-        -e "s#%CROSS_COMPILER_PREFIX%#${arche}-linux-uclibc-#" \
-        < "${ARC_GNU}"/uClibc/arc_config > .config
+    ${SED} -e "s#%KERNEL_HEADERS%#${tmp_install_dir}/include#" \
```
Comment (anthony-kolesov) [inline build-uclibc.sh:285]: Same as build-elf32.sh

File: rel-rpaths.sh [hunk 6]
```diff
@@ -39,7 +39,7 @@ fi
 # Get list of x86/x86_64 executables
 files=$(find -type f -exec file {} \; | \
     grep 'ELF 32-bit LSB executable, Intel 80386\|ELF 64-bit LSB executable, x86-64' | \
-    sed -e 's/:.*$//')
+    ${SED} -e 's/:.*$//')
```
Comment (anthony-kolesov) [inline rel-rpaths.sh:42]: No file is sourced, so $SED is unset.

File: run-uclibc-tests.sh [hunk 7]
```diff
@@ -127,7 +127,7 @@ echo "Test board:         ${ARC_TEST_BOARD_UCLIBC}"          >> ${readme}
 echo "Test IP address:    ${ARC_TEST_ADDR_UCLIBC}"           >> ${readme}
 echo "Multilib options:   ${ARC_MULTILIB_OPTIONS}"           >> ${readme}
 echo "Commfile contents:"                                    >> ${readme}
-sed < ${commfile} -e 's/^/    /'                             >> ${readme}
+${SED} < ${commfile} -e 's/^/    /'                             >> ${readme}
```
Comment (anthony-kolesov) [inline run-uclibc-tests.sh:130]: Same as build-elf32.sh

**Comment by anthony-kolesov** [comment]
Also ChangeLog entry should be added, we are still using changelogs due to historical reasons.

File: rel-rpaths.sh (hunk 6 above)
Comment (nandub) [inline rel-rpaths.sh:42]: I didn't source it here because this script is called by `build-all.sh` script.

File: build-all.sh (hunk 1 above)
Comment (nandub) [inline build-all.sh]: Good catch, didn't thought about it.

**Comment by anthony-kolesov** [comment]
@jeremybennett This looks good to me, and I've built toolchain (on Linux) with those patches. Can I merge then?

File: rel-rpaths.sh (hunk 6 above)
Comment (anthony-kolesov) [inline rel-rpaths.sh:42]: Good point. I call this file manually, that's why I've noted this. However I still has to set other environment variables before invoking this script manually, so I will just set SED as well then.

**Comment by jeremybennett** [comment]
Hi Anton,

If it passes all your tests, please merge in both 4.8 and mainline branches.

Thanks, Jeremy

**Comment by anthony-kolesov** [comment]
Merged.

Thank you, Fernando.

**Pull request closed without merging**